src/actions/actionSelectAll.ts

```diff
@@ -8,7 +8,9 @@ export const actionSelectAll = register({
       appState: {
         ...appState,
         selectedElementIds: elements.reduce((map, element) => {
-          map[element.id] = true;
+          if (!element.isDeleted) {
+            map[element.id] = true;
+          }
```

Comment:
```suggestion
          map[element.id] = !element.isDeleted;
```

Reply:
Not sure if that's what we want... but it's the same thing.

Reply:
It's not. We want to get a map of keys. We set it to `true` simply because we have to set it to something.

Reply:
But I'm not adding deleted items.. I just change the if statement without changing the logic.

Reply:
What do you mean? I'm seeing you're setting `map[element.id]` unconditionally.

Reply:
With: `!element.isDeleted;` if not deleted `true`, otherwise `false`.. we don't want to update in general?

Reply:
This action just selects elements, and replaces the `selectedElementIds` map in `appState`, which should consist only of elements being selected.

Reply:
I'm talking purely from the code point of view.. could be also be this:

```js
map[element.id] = !element.isDeleted || map[element.id];
```

Reply:
But setting something to `true/false` inside a single if statement can always be replaced with an assignment

Reply:
```
map[element.id] = !element.isDeleted || map[element.id];
```

sets the element to either `true` or `undefined`, but you'll always end up with deleted elements in the resulting map.

@dai-shi only adds the element to the map if the element isn't deleted.

Reply:
riiiiight!

Reply:
We're even :smiling_imp:

Reply:
hehe.. yes :D